editoast/src/modelsv2/scenario.rs

```diff
@@ -0,0 +1,127 @@
+use crate::error::Result;
+use crate::models::List;
+use crate::modelsv2::{Model, Ordering, Row, Tags};
+use crate::views::pagination::{Paginate, PaginatedResponse};
+use crate::views::v2::scenario::{ScenarioV2, ScenarioWithDetailsV2};
+use async_trait::async_trait;
+use chrono::NaiveDateTime;
+use diesel::sql_query;
+use diesel::sql_types::{BigInt, Text};
+use diesel::{ExpressionMethods, QueryDsl};
+use diesel_async::{AsyncPgConnection as PgConnection, RunQueryDsl};
+use editoast_derive::ModelV2;
+use serde_derive::Deserialize;
+
+#[derive(Debug, Clone, ModelV2, Deserialize)]
+#[model(table = crate::tables::scenario_v2)]
+#[model(changeset(public))]
+pub struct ScenarioModel {
+    pub id: i64,
+    pub infra_id: i64,
+    pub name: String,
+    pub description: String,
+    pub creation_date: NaiveDateTime,
+    pub last_modification: NaiveDateTime,
+    #[model(remote = "Vec<Option<String>>")]
+    pub tags: Tags,
+    pub timetable_id: i64,
+    pub study_id: i64,
+}
+
+impl From<ScenarioModel> for ScenarioV2 {
+    fn from(model: ScenarioModel) -> Self {
+        Self {
+            id: model.id,
+            infra_id: model.infra_id,
+            name: model.name,
+            description: model.description,
+            creation_date: model.creation_date,
+            last_modification: model.last_modification,
+            tags: model.tags,
+            timetable_id: model.timetable_id,
+            study_id: model.study_id,
+        }
+    }
+}
+
+#[derive(Debug, Clone, QueryableByName)]
+pub struct ScenarioWithDetailsModel {
```

Comment:
This is not a model, it's a view response. I'd suggest only keeping `views::ScenarioWithDetailsV2` and implement:

```rust
// in models
impl Scenario {
    fn infra_name(&self, conn) -> String
    fn train_count(&self, conn) -> usize
}

// in views
impl ScenarioWithDetailsV2 {
    fn from_scenario(conn, scenario: &Scenario) -> Self {
        Self {
            infra_name: scenario.infra_name(conn),
            train_count: scenario.train_count(conn),
            scenario
        }
    }
}
```